src/regfox/regfox_api.js

```diff
@@ -79,15 +81,45 @@ const login = async (email, password) => {
   }).then(response => response.json())
     .then(response => {
       if (get(response, 'data.mutationResponse.errors')) {
-        throw new Error(response);
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
       }
 
       if (!get(response, 'data.mutationResponse.success', false)) {
-        throw new Error(response);
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
       }
 
       return response.data.mutationResponse;
     });
 }
 
-export { searchRegistrations, exchangeBearerToken, login, REGFOX_GRAPHQL_URL, REGFOX_EXCHANGE_TOKEN_URL };
\ No newline at end of file
+/**
+ * Returns a nicely parsed object containing registration info given an id.
+ * Returns a promise Error if the network is down or parsing failed.
+ * 
+ * @param {*} id of a registration, note this is **not** the RegistrationId, its just the id
+ */
+const getRegistrationInfo = async (id) => {
+  const fullUrl = getRegistrationInfoUrl(id);
+  return fetch(fullUrl, {
+    method: 'GET',
+    headers: buildHeaders(undefined)
+  }).then(response => response.json())
+    .then(response => {
+      if (get(response, 'error')) {
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
+      }
+
+      if (!get(response, 'data')) {
+        throw new Error(`Could not parse or process response, see details to fix here: ${JSON.stringify(response)}`);
+      }
+
+      return parseRegfoxGetRegistrationResponse(response.data);
+    });
+}
+
+// Only exported for testing, DONT CALL THIS METHOD!
+const getRegistrationInfoUrl = (id) => {
+  return REGFOX_GET_REGISTRATION_URL.replace('${id}', id);
+}
+
+export { getRegistrationInfo, searchRegistrations, exchangeBearerToken, login, REGFOX_GRAPHQL_URL, REGFOX_EXCHANGE_TOKEN_URL, getRegistrationInfoUrl };
```

Comment:
Thoughts on my `export <definition>` syntax vs this auto-suggested block format? Not a concern, just curious if you have an opinion.

Reply:
I'm slightly partial towards the export being at the bottom for two reasons
1. That's how old javascript did it :P
2. Keeps all of the exports in one single location

However, if we are only exporting a single method/class from a file I'm happy to use `export <definition>` instead.